src/hangar/backends/hdf5.py

```diff
@@ -0,0 +1,552 @@
+    write to the same dataset schema.
+    '''
+
+    def __init__(self, repo_path: os.PathLike, schema_shape: tuple, schema_dtype: np.dtype):
+        self.repo_path = repo_path
+        self.schema_shape = schema_shape
+        self.schema_dtype = schema_dtype
+
+        self.rFp: MutableMapping[str, h5py.File] = {}
+        self.wFp: MutableMapping[str, h5py.File] = {}
+        self.Fp = ChainMap(self.rFp, self.wFp)
+
+        self.mode: str = None
+        self.hIdx: int = None
+        self.w_uid: str = None
+        self.hMaxSize: int = None
+        self.hNextPath: int = None
+        self.hColsRemain: int = None
+
+        self.slcExpr = np.s_
+        self.slcExpr.maketuple = False
+        self.fmtParser = HDF5_00_Parser()
+
+        self.STAGEDIR = pjoin(self.repo_path, c.DIR_DATA_STAGE, self.fmtParser.FmtCode)
+        self.REMOTEDIR = pjoin(self.repo_path, c.DIR_DATA_REMOTE, self.fmtParser.FmtCode)
+        self.DATADIR = pjoin(self.repo_path, c.DIR_DATA, self.fmtParser.FmtCode)
+        self.STOREDIR = pjoin(self.repo_path, c.DIR_DATA_STORE, self.fmtParser.FmtCode)
+        if not os.path.isdir(self.DATADIR):
+            os.makedirs(self.DATADIR)
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *exc):
+        if self.w_uid in self.wFp:
+            self.wFp[self.w_uid]['/'].attrs.modify('next_location', (self.hNextPath, self.hIdx))
+            self.wFp[self.w_uid]['/'].attrs.modify('collections_remaining', self.hColsRemain)
+            self.wFp[self.w_uid].flush()
+
+    def open(self, mode: str, *, remote_operation: bool = False):
+        '''Open an hdf5 file handle in the Handler Singleton
+
+        Parameters
+        ----------
+        mode : str
+            one of `r` or `a` for read only / read-write.
+        repote_operation : optional, kwarg only, bool
+            if this hdf5 data is being created from a remote fetch operation, then
+            we don't open any files for reading, and only open files for writing
+            which exist in the remote data dir. (default is false, which means that
+            write operations use the stage data dir and read operations use data store
+            dir)
+        '''
+        self.mode = mode
+        if self.mode == 'a':
+            process_dir = self.REMOTEDIR if remote_operation else self.STAGEDIR
+            if not os.path.isdir(process_dir):
+                os.makedirs(process_dir)
+
+            process_uids = [psplitext(x)[0] for x in os.listdir(process_dir) if x.endswith('.hdf5')]
+            for uid in process_uids:
+                file_pth = pjoin(process_dir, f'{uid}.hdf5')
+                self.rFp[uid] = partial(h5py.File, file_pth, 'r', swmr=True, libver='latest')
+
+        if not remote_operation:
+            if not os.path.isdir(self.STOREDIR):
+                return
+            store_uids = [psplitext(x)[0] for x in os.listdir(self.STOREDIR) if x.endswith('.hdf5')]
```

Comment:
Technically yes, but doing so would require that we persist the `.hdf5` file extension in every data lmdb record (to avoid the for the following `os.path.splitext` function call) However, this would waste disk space (compounded for every array record stored in the database via that backend). 

It's not an issue, since the only files which should ever be placed in this directory are the `.hdf5` files created by hangar. We just need the checks in order to exclude OS created `"."` (`dot`)  files/directories such as MacOSX resource forks... if the user modifies the contents of this directory, the local copy of that repo would just be crazily screwed up. (think what would happen if you just went about deleting or adding files to a `.git` directory).